lib/trino-filesystem/src/main/java/io/trino/filesystem/cache/CacheInputFile.java

```diff
@@ -30,12 +31,14 @@ public final class CacheInputFile
     private final TrinoInputFile delegate;
     private final TrinoFileSystemCache cache;
     private final CacheKeyProvider keyProvider;
+    private OptionalLong length;
```

Comment:
there are not thread safe, right?

Reply:
i think so, there's no threading related code in any other implementation